File: basic-python/1_reverse_a_string.py
#REVERSE A STRING
# Reverse the provided string.
# You may need to turn the string into an array before you can reverse it.
# Your result must be a string.

def reverseString(str):
    return str[::-1] #This is extended slice syntax. It works by doing [begin:end:step] -
    #by leaving begin and end off and specifying a step of -1, it reverses a string.

# OR
def reverseString(str):
    return ''.join(reversed(str))
# TESTS
print (reverseString("hello")) #should return a string.
print (reverseString("hello")) #should become "olleh".
print (reverseString("Howdy")) #should become "ydwoH".
print (reverseString("Greetings from Earth")) #should return "htraE morf sgniteerG"

File: basic/1_reverse_a_string.js
//REVERSE A string

//Reverse the provided string.
// You may need to turn the string into an array before you can reverse it.
// Your result must be a string.

function reverseString(str) {
  return str
    .split('')
    .reverse()
    .join('');
}

//OR
function reverseString(str) {
  var array = str.split('');
  var arrrev = [];
  var revstr = '';

  for (var i = array.length - 1; i >= 0; i--) {
    revstr += array[i];
  }
  return revstr;
}

//TESTS

reverseString('hello'); //should return a string.
reverseString('hello'); //should become "olleh".
reverseString('Howdy'); //should become "ydwoH".
reverseString('Greetings from Earth'); //should return "htraE morf sgniteerG".
